### PR TITLE
(1) Adds Variable option to generate method

### PR DESCRIPTION
**Pull request by someshkoli**
This option allows user to provide postman variales for sdk generation. User can provide multiple variable list (global/env) and final variable will be resolved according to postman variable scope resolution.

    variables: {
        var1: "val1",
        var2: "val2"
    }
  
Removes codeOption and passes variableList through sdkOptions

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
   fs = require('fs'),
   path = require('path'),
+  _ = require('lodash'),
   fetch = require('node-fetch'),
-  _ = require('lodash');
+  languageMap = require('./assets/languageMap.json'),
+  generator = require('../sdkgen/nodejs-request').generate; // TODO fix language mapping. this is temporary
 var codeOptions = {
-  SDK_enabled: true
+  SDKGEN_enabled: true
```
Comment (umeshp7) [inline lib/index.js]: Where is this used?

File: lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * @param {Object} variableLists - Array of PostmanVariableList to be combined together
+ * @returns {PostmabVariableList} - PostmanVariableList instance containing variables of all variableLists provided
+ */
+function resolveVariables (variableLists) {
+  let variables = {},
+    variableListArray = [],
+    variableList = new sdk.VariableList();
+
+  if (variableLists.global) {
```
Comment (umeshp7) [inline lib/utils.js]: Let's not have globals. environments, collection etc. Lets assume we are getting a unified list directly

File: sdkgen/nodejs-request/lib/index.js [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
+const processCollection = require('./util').processCollection,
+  sanitize = require('./util').sanitize,
+  combineVariablesLists = require('../../../lib/utils').combineVariablesLists;
 
 /**
  * Generates sdk for nodejs-request
 
  * @param {PostmanCollection} collection - Postman collection Instance
+ * @param {PostmamVariableList} variables - Postman Variable List with resolved scope
  * @param {Object} options - postman-code-generators options
```
Comment (umeshp7) [inline sdkgen/nodejs-request/lib/index.js]: This generate function should not take p-c-g options. 
You are the consumer of that module, you will decide what options has to be used.

File: sdkgen/nodejs-request/lib/index.js [hunk 4]
```diff
@@ -1,21 +1,23 @@
...
  * @returns {String} - sdk snippet for input collection
  * TODO add indentation to entire snippet
  */
-function generate (collection, options, callback) {
+function generate (collection, variables, options, callback) {
   var snippet = '',
     indent = options.indentType === 'Tab' ? '\t' : ' ';
   indent = indent.repeat(options.indentCount);
+
   processCollection(collection, options, (err, collectionSnippet) => {
```
Comment (umeshp7) [inline sdkgen/nodejs-request/lib/index.js:20]: processCollection doesn't have a callback right?

File: lib/index.js [hunk 5]
```diff
@@ -51,6 +51,12 @@ function getSDKOptions () {
       type: Boolean,
       default: false,
       description: 'Specifies to add or not to add README along with sdk'
+    },
+    variables: {
```
Comment (umeshp7) [inline lib/index.js:52]: Where are we passing variables exactly? Here or through the generate function?

**Review by umeshp7 — CHANGES_REQUESTED** [review]

File: lib/index.js (hunk 5 above)
Comment (someshkoli) [reply lib/index.js:52]: these are the variables which the user will provide

File: sdkgen/nodejs-request/lib/index.js (hunk 4 above)
Comment (someshkoli) [reply sdkgen/nodejs-request/lib/index.js:20]: In this version it has. This thing has been revised in #3

**Pull request merged**